extension/src/background/helpers/dataStorage.ts

```diff
@@ -147,14 +148,50 @@ export const migrateTokenIdList = async () => {
     };
     await localStore.setItem(TOKEN_ID_LIST, newTokenList);
   }
-  await migrateDataStorageVersion();
+  await migrateDataStorageVersion("1.0.0");
+};
+
+const migrateTestnetSorobanRpcUrlNetworkDetails = async () => {
+  const localStore = dataStorageAccess(browserLocalStorage);
+  const storageVersion = (await localStore.getItem(STORAGE_VERSION)) as string;
+
+  if (!storageVersion || semver.lt(storageVersion, "2.0.0")) {
+    const networksList: NetworkDetails[] =
+      (await localStore.getItem(NETWORKS_LIST_ID)) || DEFAULT_NETWORKS;
+
+    const migratedNetworkList = networksList.map((network) => {
+      if (network.network === NETWORKS.TESTNET) {
+        return {
+          ...TESTNET_NETWORK_DETAILS,
+          sorobanRpcUrl: SOROBAN_RPC_URLS[NETWORKS.TESTNET],
+        };
+      }
+
+      return network;
+    });
+
+    const currentNetwork = await localStore.getItem(NETWORK_ID);
+
+    if (currentNetwork.network === NETWORKS.TESTNET) {
+      await localStore.setItem(NETWORK_ID, TESTNET_NETWORK_DETAILS);
+    }
+
+    await localStore.setItem(NETWORKS_LIST_ID, migratedNetworkList);
+    await migrateDataStorageVersion("2.0.0");
+  }
+};
+
+export const versionedMigration = async () => {
+  // sequentially call migrations in order to enforce smooth schema upgrades
+
+  await migrateTokenIdList();
+  await migrateTestnetSorobanRpcUrlNetworkDetails();
```

Comment:
the idea here is to make sure people who are multiple migrations behind go through each migration iteratively. jumping ahead to a migration script they're not yet ready for could cause big issues.

I def still need to do some testing on this

Reply:
oh interesting yeah I guess all of the listeners aren't really sequential.

Reply:
Yeah, I haven't really dug into how they work, but from testing, I've observed that they don't guarantee sequence. Assuming they all just work async 🤷

Reply:
for sure, I guess that makes sense with an event loop. This makes sense to me at a glance, even if the events are async, if we only have one "migration" event listener that sequentially runs the migrations then that should enforce order... _I think_